Engine/DataFeeds/Enumerators/Factories/BaseDataSubscriptionEnumeratorFactory.cs

```diff
@@ -51,15 +66,36 @@ public IEnumerator<BaseData> CreateEnumerator(SubscriptionRequest request, IData
         {
             var sourceFactory = (BaseData)Activator.CreateInstance(request.Configuration.Type);
 
-            return (
-                from date in _tradableDaysProvider(request)
-                let source = sourceFactory.GetSource(request.Configuration, date, false)
-                let factory = SubscriptionDataSourceReader.ForSource(source, dataFileProvider, request.Configuration, date, false)
-                let entriesForDate = factory.Read(source)
-                from entry in entriesForDate
-                select entry
-                )
-                .GetEnumerator();
+            foreach (var date in _tradableDaysProvider(request))
+            {
+                var currentSymbol = request.Configuration.MappedSymbol;
+                request.Configuration.MappedSymbol = GetMappedSymbol(request, date);
+                var source = sourceFactory.GetSource(request.Configuration, date, false);
+                request.Configuration.MappedSymbol = currentSymbol;
+                var factory = SubscriptionDataSourceReader.ForSource(source, dataFileProvider, request.Configuration, date, false);
+                var entriesForDate = factory.Read(source);
+                foreach(var entry in entriesForDate)
+                {
+                    yield return entry;
+                }
+            }
+        }
+        private string GetMappedSymbol(SubscriptionRequest request, DateTime date)
+        {
+            var config = request.Configuration;
+            if (config.Symbol.ID.SecurityType == SecurityType.Option ||
+                config.Symbol.ID.SecurityType == SecurityType.Equity )
+            {
+                var mapFile = config.Symbol.HasUnderlying ?
+                        _mapFileResolver.ResolveMapFile(config.Symbol.Underlying.ID.Symbol, config.Symbol.Underlying.ID.Date) :
+                        _mapFileResolver.ResolveMapFile(config.Symbol.ID.Symbol, config.Symbol.ID.Date);
```

Comment:
I think I remember seeing this somewhere else. Perhaps this can be encapsulated/refactored into a method on `MapFileResolver` ?